frontend/packages/console-app/src/hooks/useCSPVioliationDetector.tsx

```diff
@@ -154,49 +155,47 @@ const useCSPViolationReporter: UseCSPVilationReporter = () => {
         }
       }
     },
-    [fireTelemetryEvent, getRecords, t, toastContext, updateRecords],
-  );
-};
-
-export const useCSPViolationDetector = () => {
-  const reportViolation = useCSPViolationReporter();
-  const onCSPViolation = React.useCallback(
-    (event) => {
-      // Attempt to infer Console plugin name from SecurityPolicyViolation event
-      const pluginName =
-        getPluginNameFromResourceURL(event.blockedURI) ||
-        getPluginNameFromResourceURL(event.sourceFile);
-
-      reportViolation(pluginName, event);
-    },
-    [reportViolation],
+    [fireTelemetryEvent, getRecords, t, toastContext, updateRecords, pluginStore],
   );
 
   React.useEffect(() => {
-    document.addEventListener('securitypolicyviolation', onCSPViolation);
+    document.addEventListener('securitypolicyviolation', reportViolation);
     return () => {
-      document.removeEventListener('securitypolicyviolation', onCSPViolation);
+      document.removeEventListener('securitypolicyviolation', reportViolation);
     };
-  }, [onCSPViolation]);
+  }, [reportViolation]);
 };
 
+// A subset of properties from a SecurityPolicyViolationEvent which identify a unique CSP violation
```

Comment:
just a nitpick but this should use jsdoc format for better Intellisense integration